src/recordField.ts

```diff
@@ -0,0 +1,85 @@
+import { Dictionary, Documented, expandUrl, loadField, LoadingOptions, Saveable, ValidationException, typedslUnionOfPrimitiveTypeLoaderOrRecordSchemaLoaderOrEnumSchemaLoaderOrArraySchemaLoaderOrStrtypeOrArrayOfUnionOfPrimitiveTypeLoaderOrRecordSchemaLoaderOrEnumSchemaLoaderOrArraySchemaLoaderOrStrtype2, unionOfNoneTypeOrStrTypeOrArrayOfStrType, uriStrtypeTrueFalseNone } from './util/internal'
+
+export class RecordField extends Documented {
+  name: any
+  type: any
+  doc?: any
+  extensionFields?: Dictionary<any>
+  loadingOptions?: LoadingOptions
+
+  constructor (name: any, type: any, doc?: any, extensionFields?: Dictionary<any>, loadingOptions?: LoadingOptions) {
+    super()
+    this.name = name
+    this.type = type
+    this.doc = doc
+    this.extensionFields = extensionFields ?? {}
+    this.loadingOptions = loadingOptions ?? new LoadingOptions({})
+  }
+
+  static override async fromDoc (doc: any, baseuri: string, loadingOptions: LoadingOptions, docRoot?: string): Promise<Saveable> {
+    const _doc = Object.assign({}, doc)
+    const errors: ValidationException[] = []
+    if ('name' in _doc) {
+      try {
+        var name: string|undefined = await loadField(_doc.name, uriStrtypeTrueFalseNone, baseuri, loadingOptions)
+      } catch (e) {
+        if (e instanceof ValidationException) {
+          errors.push(new ValidationException('the `name` field is not valid because: ', [e]))
+        }
+      }
+    }
+
+    const originalNameIsUndefined = (name === undefined)
+    if (originalNameIsUndefined) {
+      if (docRoot != null) {
+        name = docRoot
+      } else {
+        throw new ValidationException('Missing name')
+      }
+    } else {
+      baseuri = name as string
+    }
+
+    if ('doc' in _doc) {
+      try {
+        doc = await loadField(_doc.doc, unionOfNoneTypeOrStrTypeOrArrayOfStrType, baseuri, loadingOptions)
+      } catch (e) {
+        if (e instanceof ValidationException) {
+          errors.push(new ValidationException('the `doc` field is not valid because:', [e]))
+        }
+      }
+    } else {
+      doc = undefined
+    }
+    let type
+    try {
+      type = await loadField(_doc.type, typedslUnionOfPrimitiveTypeLoaderOrRecordSchemaLoaderOrEnumSchemaLoaderOrArraySchemaLoaderOrStrtypeOrArrayOfUnionOfPrimitiveTypeLoaderOrRecordSchemaLoaderOrEnumSchemaLoaderOrArraySchemaLoaderOrStrtype2, baseuri, loadingOptions)
+    } catch (e) {
+      if (e instanceof ValidationException) {
+        errors.push(new ValidationException('The `type` field is not valid because: ', [e]))
```

Comment:
Most messages that start with `the` and talk about a field start with a lower case vowel. Maybe change it here to be uniform? (sorry my OCD :laughing: )

Reply:
Should definitely be uniform 😅